### DISCUSSION TITLE
Tenant Console Loading Forever

### DISCUSSION THREAD

**Discussion by Sunilsai1066** [minio/minio#20210] [Q&A]
## Expected Behavior
Console should load all the objects properly

## Current Behavior
Tenant console isn't able to load objects in bucket.

## Steps to Reproduce (for bugs)
Deploy helm chart of operator of version v5.0.12. 
Enable ingress in values yaml file

## Context
I've deployed operator helm chart in my kubernetes cluster and enabled ingress section in values yaml file to true. Now it's not loading objects in bucket in tenant console. If i do port forward it's working fine without any issue, but issue is occuring when ingress is enabled.

## Your Environment

* Version used (`minio --version`): MinIO Operator v5.0.12 helm chart (Same issue with v5.0.11 too)
* Server setup and configuration: Ingress Config in values yaml file

>  ingress:
>     enabled: true
>     ingressClassName: nginx
>     labels: { }
>     annotations: {}
>     tls:
>       - hosts:
>           - minio-test.com
>         secretName: minio-cred
>     host: minio-test.com
>     path: /
>     pathType: Prefix

I also tried adding some annotations in values yaml file after seeing some suggestions to add websocket support like below, Still same issue

>     annotations:
>       kubernetes.io/ingress.class: nginx
>       nginx.ingress.kubernetes.io/proxy-read-timeout: "7200"
>       nginx.ingress.kubernetes.io/proxy-send-timeout: "7200"
>       nginx.ingress.kubernetes.io/server-snippets: |
>         location / {
>         proxy_set_header Host $http_host;
>         proxy_set_header X-Real-IP $remote_addr;
>         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
>         proxy_set_header X-Forwarded-Proto $scheme;
>         
>         proxy_connect_timeout 300;
>         # Default is HTTP/1, keepalive is only enabled in HTTP/1.1
>         proxy_http_version 1.1;
>         proxy_set_header Connection "";
>         chunked_transfer_encoding off;
>         proxy_pass https://minio-test.com;
>         }

* Operating System and version (`uname -a`):

**Comment by harshavardhana** [comment]
> annotations:
>   kubernetes.io/ingress.class: nginx
>   nginx.ingress.kubernetes.io/proxy-read-timeout: "7200"
>   nginx.ingress.kubernetes.io/proxy-send-timeout: "7200"
>   nginx.ingress.kubernetes.io/server-snippets: |
>     location / {
>     proxy_set_header Host $http_host;
>     proxy_set_header X-Real-IP $remote_addr;
>     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
>     proxy_set_header X-Forwarded-Proto $scheme;
>     
>     proxy_connect_timeout 300;
>     # Default is HTTP/1, keepalive is only enabled in HTTP/1.1
>     proxy_http_version 1.1;
>     proxy_set_header Connection "";
>     chunked_transfer_encoding off;
>     proxy_pass https://minio-test.com;
>     }

where is the Console UI port being exposed? I see that you have one layer of ingress pointing to minio-test.com -is this domain resolvable at ingress layer?

Looks to me like a configuration issue. Moving this to discussion.

**Reply by Sunilsai1066** [reply]
Yes this domain is resolvable. Like i mentioned Console is working fine till i go to object browser page. There it's struck which loading objects in bucket. I'm able to login, create tenants, create buckets and all that through UI.

Regarding where Console UI is exposed, i enabled ingress stanza on values yaml file
https://github.com/minio/operator/blob/v5.0.12/helm/operator/values.yaml#L283

**Reply by harshavardhana** [reply]
you need to make sure to enable websocket support 

```
      # To support websockets in MinIO versions released after January 2023
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      # Some environments may encounter CORS errors (Kubernetes + Nginx Ingress)
      # Uncomment the following line to set the Origin request to an empty string
      # proxy_set_header Origin '';

      chunked_transfer_encoding off;
```

**Reply by Sunilsai1066** [reply]
Below is my changes in ingress section in values yaml file. I still see the loading issue.

```
  ingress:
    enabled: true
    ingressClassName: nginx
    labels: { }
    annotations:
    nginx.ingress.kubernetes.io/configuration-snippet: |
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      chunked_transfer_encoding off;
    tls:
      - hosts:
        - minio-test.com
        secretName: minio-cred
    host: minio-test.com
    path: /
    pathType: Prefix
```
